plugin/src/main/java/com/gtnewhorizons/retrofuturagradle/modutils/MigrateMappingsTask.java

```diff
@@ -0,0 +1,194 @@
+package com.gtnewhorizons.retrofuturagradle.modutils;
+
+import com.gtnewhorizons.retrofuturagradle.mcp.GenSrgMappingsTask;
+import com.gtnewhorizons.retrofuturagradle.mcp.InjectTagsTask;
+import com.gtnewhorizons.retrofuturagradle.mcp.SharedMCPTasks;
+import com.gtnewhorizons.retrofuturagradle.util.Utilities;
+import com.opencsv.CSVReader;
+import org.apache.commons.io.FileUtils;
+import org.cadixdev.lorenz.MappingSet;
+import org.cadixdev.lorenz.io.MappingFormats;
+import org.cadixdev.lorenz.io.MappingsWriter;
+import org.cadixdev.lorenz.model.ClassMapping;
+import org.cadixdev.lorenz.model.FieldMapping;
+import org.cadixdev.lorenz.model.InnerClassMapping;
+import org.cadixdev.lorenz.model.MethodMapping;
+import org.cadixdev.lorenz.model.TopLevelClassMapping;
+import org.cadixdev.mercury.Mercury;
+import org.cadixdev.mercury.mixin.MixinRemapper;
+import org.cadixdev.mercury.remapper.MercuryRemapper;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.plugins.JavaPluginExtension;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.jvm.tasks.Jar;
+import org.gradle.api.tasks.options.Option;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class MigrateMappingsTask extends DefaultTask {
+
+    private static final boolean DEBUG_WRITE_DIFF = Boolean.parseBoolean(System.getenv("RFG_DEBUG_WRITE_MAPPING_DIFF"));
+
+    private File mcpDir;
```

Comment:
These should probably be abstract DirectoryProperty returning getter functions, to use gradle's DI to generate them. This will autowire task dependencies and a few other features, see how I use it in other tasks.